app/models/partners/profile.rb

```diff
@@ -108,5 +115,26 @@ class Profile < Base
     def has_no_social_media?
       website.blank? && twitter.blank? && facebook.blank? && instagram.blank?
     end
+
+    def client_share_total
+      tot = 0
+      served_areas.each do |served_area|
+        tot += served_area.client_share
+      end
+      tot
+    end
+
+    def client_share_is_0_or_100
+      # business logic:  the client share has to be 0 or 100 -- although it is an estimate only,  making it 0 (not
+      # specified at all) or 100 means we won't have people overallocating (> 100) and that they think about what
+      # their allocation actually is
+      value = client_share_total
+      check = (value == 0 || value == 100)
+      if !check
+
+        errors.add(:base, "Total client share must be 0 or 100")
+      end
+      check
+    end
```

Comment:
This can be tidied up:

```ruby
def client_share_is_0_or_100
  value = client_share_total
  if value != 0 && value != 100
    errors.add(:base, "Total client share must be 0 or 100")
  end
end
```